integrasjonstest/src/test/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/integrasjonstest/TrengerIT.kt

```diff
@@ -1,98 +1,105 @@
-            )
+            Key.EVENT_NAME to EventName.TRENGER_REQUESTED.toJson(),
+            Key.DATA to "".toJson(),
+            Key.UUID to transactionId.toJson(),
+            DataFelt.FORESPOERSEL_SVAR to mockTrengerInntekt().toJson(TrengerInntekt.serializer())
         )
 
-        Thread.sleep(12000)
+        waitForMessages(12000)
 
-        with(filter(EventName.TRENGER_REQUESTED, behovType = BehovType.HENT_TRENGER_IM).first()) {
-            // Ble lagret i databasen
-            Assertions.assertEquals(transactionID, this[Key.UUID.str].asText())
-        }
+        messages.filter(EventName.TRENGER_REQUESTED)
+            .filter(BehovType.HENT_TRENGER_IM, loesningPaakrevd = false)
+            .first()
+            .fromJsonMapOnlyKeys()
+            .let {
+                // Ble lagret i databasen
+                it[Key.UUID]?.fromJson(UuidSerializer) shouldBe transactionId
+            }
 
-        with(filter(EventName.TRENGER_REQUESTED, behovType = BehovType.VIRKSOMHET).first()) {
-            // Ble lagret i databasen
-            Assertions.assertEquals(transactionID, this[Key.UUID.str].asText())
-        }
+        messages.filter(EventName.TRENGER_REQUESTED)
+            .filter(BehovType.VIRKSOMHET, loesningPaakrevd = false)
+            .first()
+            .fromJsonMapOnlyKeys()
+            .let {
+                // Ble lagret i databasen
+                it[Key.UUID]?.fromJson(UuidSerializer) shouldBe transactionId
+            }
 
-        with(filter(EventName.TRENGER_REQUESTED, behovType = BehovType.FULLT_NAVN).first()) {
-            // Ble lagret i databasen
-            Assertions.assertEquals(transactionID, this[Key.UUID.str].asText())
-        }
+        messages.filter(EventName.TRENGER_REQUESTED)
+            .filter(BehovType.FULLT_NAVN, loesningPaakrevd = false)
+            .first()
+            .fromJsonMapOnlyKeys()
+            .let {
+                // Ble lagret i databasen
+                it[Key.UUID]?.fromJson(UuidSerializer) shouldBe transactionId
+            }
 
-        with(filter(EventName.TRENGER_REQUESTED, behovType = BehovType.INNTEKT).first()) {
-            // Ble lagret i databasen
-            Assertions.assertEquals(transactionID, this[Key.UUID.str].asText())
-        }
+        messages.filter(EventName.TRENGER_REQUESTED)
+            .filter(BehovType.INNTEKT, loesningPaakrevd = false)
+            .first()
+            .fromJsonMapOnlyKeys()
+            .let {
+                // Ble lagret i databasen
+                it[Key.UUID]?.fromJson(UuidSerializer) shouldBe transactionId
+            }
 
-        val trengerResultatJson = redisStore.get(RedisKey.of(CLIENT_ID))
+        val trengerResultatJson = redisStore.get(RedisKey.of(Mock.clientId.toString()))
         println("In test $trengerResultatJson")
         val objekt = trengerResultatJson?.fromJson(TrengerData.serializer())
         println(objekt)
     }
+
+    private object Mock {
```

Comment:
I am not fan of the main mocking parameters to be in a nested class at the bottom of the test. When I open the test I want these values to be almost the first thing I am seeing. So this kind of private Mock object is a bit unnessesary for me and also makes it less readable for me.

Reply:
Jeg foretrekker det nederst, men har ikke sterke meninger om det. Vi kan flytte det opp. så lenge vi er konsekvente.

Jeg foretrekker dog at vi beholder det i den logiske samlingen som objektet gir.